docs/src/DeveloperDocumentation/caching.md

```diff
@@ -0,0 +1,95 @@
+```@meta
+CurrentModule = Oscar
+```
+
+# Caching parent objects in OSCAR
+
+Many functions in OSCAR that construct parent objects (such as rings, modules,
+groups, etc.) have an optional keyword argument `cached::Bool`. If set to
+`true` then the object is put into a cache, and when the construction function
+is later called again with identical inputs, then the cached object is
+returned instead of creating a new object. In contrast when `cached` is set to
+`false` then each time a new object is returned.
+
+Example:
+```jldoctest
+julia> R1, = polynomial_ring(QQ, :x; cached = true);
+
+julia> R2, = polynomial_ring(QQ, :x; cached = true);
+
+julia> R1 === R2  # identical as both were created with `cached = true`
+true
+
+julia> R3, = polynomial_ring(QQ, :x; cached = false);
+
+julia> R1 === R3  # not identical as R3 was created with `cached = false`
+false
+
+julia> R4, = polynomial_ring(QQ, :y; cached = true);
+
+julia> R1 === R4  # not identical despite `cached = true` due to differing variable names
+false
+```
+
+## Why cache parent objects?
+
+The main reason for supporting caching of parent objects is **user convenience**:
+experience shows that most mathematicians (espescially those who are not also
+programmers; but it really affects all) are surprised if, say, `QQ[:x] == Q[:x]`
+produces `false`.
+
+For interactive use, it is often simply convenient: e.g. taking the characteristic
+polynomials of two matrices and multiplying them, without worrying about specifying
+the parent rings of the polynomials.
```

Comment:
Funnily enough, exactly this example doesn't work because caching is deliberately turned off:
```
julia> M = QQ[1 0; 0 1];

julia> charpoly(M)*charpoly(M)
ERROR: Incompatible polynomial rings in polynomial operation
[...]
```

Reply:
Ouch! Thank you very much for fact checking this.

That makes me question the point of entire caching thing even more *sigh*.

So, does anyone then have a not completely unconvincing example for a situation where a cached parent helps a "naive" user?